addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/httputils/HtmlContextAnalyserUnitTest.java

```diff
@@ -0,0 +1,98 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrules.httputils;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import java.util.HashMap;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.testutils.TestUtils;
+
+public class HtmlContextAnalyserUnitTest extends TestUtils {
+    private HttpMessage msg;
+
+    @Test
+    void shouldGetCorrectParentTag() throws Exception {
+        msg = new HttpMessage();
+        msg.setRequestHeader("GET /index.html HTTP/1.1");
+        msg.setResponseBody("<html><body alert=\"\"></body></html>");
+        HtmlContextAnalyser analyser = new HtmlContextAnalyser(msg);
+        String catcher = "alert";
+        List<HtmlContext> contexts = analyser.getHtmlContexts(catcher, null, 0);
+        assertThat(contexts.size(), is(equalTo(1)));
+        for (HtmlContext ctx : contexts) {
+            assertThat(ctx.getParentTag(), is(equalTo("body")));
+        }
+    }
+
+    @Test
+    void shouldGetAtrributeNamesAndValues() throws Exception {
+        msg = new HttpMessage();
+        msg.setRequestHeader("GET /index.html HTTP/1.1");
+        msg.setResponseBody("<html><body assert = \"x\" onclick = \"alert('1')\"></body></html>");
+        HtmlContextAnalyser analyser = new HtmlContextAnalyser(msg);
+        String catcher = "assert = \"x\" onclick = \"alert('1')\"";
+        List<HtmlContext> contexts = analyser.getHtmlContexts(catcher, null, 0);
+        assertThat(contexts.size(), is(equalTo(1)));
+        for (HtmlContext ctx : contexts) {
+            HashMap<String, String> attrMap = ctx.getTagAttributes();
+            assertThat(attrMap.get("assert"), is(equalTo("x")));
+            assertThat(attrMap.get("onclick"), is(equalTo("alert('1')")));
+        }
+    }
+
+    @Test
+    void shouldGetAttributeName() throws Exception {
+        msg = new HttpMessage();
+        msg.setRequestHeader("GET /index.html HTTP/1.1");
+        msg.setResponseBody("<html><body assert=\"\"></body></html>");
+        HtmlContextAnalyser analyser = new HtmlContextAnalyser(msg);
+        String catcher = "assert";
+        List<HtmlContext> contexts = analyser.getHtmlContexts(catcher, null, 0);
+        assertThat(contexts.size(), is(equalTo(1)));
+        for (HtmlContext ctx : contexts) {
+            HashMap<String, String> attrMap = ctx.getTagAttributes();
+            assertThat(ctx.getParentTag(), is(equalTo("body")));
+            assertThat(attrMap.get("assert"), is(equalTo("")));
+        }
+    }
+
+    @Test
+    void shouldGetParsedAttributes() {
+        String attr = " onclick = \"alert(100)\" accesskey = \"x\" n=\"\"";
```

Comment:
I'd like to see some tests, esp with "invalid" HTML. I dont expect the code to find vulns in invalid HTML, I'd just like to see it not throwing any exceptions. Browsers handle all sorts of bad HTML so we're bound to come across it and it would be a pain if this code threw exceptions everytime it hit something out of the ordinary :)